Codereview of 4 from 17

thomasporro opened the pull request:
Hi Vlad,
here the class i've been developing the last two weeks. I wrote/helped writing all the methods where my name is on the author section. I like to write a lot of comment to explain what my code is doing, so tell me if there are too many comments. Sorry if the code review arrived so late but in the last days I was really busy.
Thank you for your time!

Edit: I had accidentaly commited some extra file but now i've deleted them. By the way the class of the review is ImageProcessing.java

PietroPrandini committed:
Initial commit

PietroPrandini committed:
Dir

PietroPrandini committed:
removeMe.md added to preserve the directories structures on github

PietroPrandini committed:
README deleted

PietroPrandini committed:
.gitignore updated with the version contained in the CommonDemo branch

PietroPrandini committed:
Gradle files

thomasporro committed:
Create ImageProcessing.java

thomasporro committed:
Update ImageProcessing.java

thomasporro committed:
Update ImageProcessing.java

thomasporro committed:
Delete .gitignore

thomasporro committed:
Delete build.gradle

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete settings.gradle

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete gradle-wrapper.properties

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete removeMe.md

thomasporro committed:
Delete build.gradle

thomasporro committed:
Delete gradle.properties

MrVlado commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+        //Invert the colors of "img" onto itself
+        Core.bitwise_not(img, img);
+
+        //Detect the edges in the image
```
//It isn't clear what the variables declared below mean. Like threshold,aperture size and 12gradient

thomasporro replied:
You can find the meaning of the variables in the documentation that i linked at the beginning of the class

MrVlado commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+        double threshold2 = 200;
+        int apertureSize = 3;
+        boolean l2gradient = false;
+        Imgproc.Canny(img, img, threshold1, threshold2, apertureSize, l2gradient);
```
I Don't get exactly what this method does

thomasporro replied:
It detects the edges of the image, in case of any doubt the documentation is linked at the beginning of the class

MrVlado commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+        //Turns the image in grayscale and put it in a matrix
+        Log.d(TAG, "Image path = " + imagePath);
+        Mat img = conversion(imagePath);
+        //save(img, "grayScale.jpg");
```
Why is this line commented ? If you used it in the debugging phase what was his scope?

MrVlado replied:
Just got to the end of file, now looks clear.

thomasporro replied:
I will move tha main comment here

MrVlado commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+
+        //Transforms a grayscale image to a binary image using the gaussian algorithm
+        Mat threshold = new Mat();
+        double maxValue = 200;
```
What does this variable mean ?

thomasporro replied:
This variable are the value of the threshold to apply, in case of any doubt the documentation is linked at the beginning of the class

MrVlado commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+     * @throws FileNotFoundException if the imagePath doesn't exist
+     * @author Oscar Garrido (g1)
+     */
+    public Mat conversion(String imagePath) throws FileNotFoundException {
```
You created a method with the same name of this one, but with a different signature. The 2 methodes do completely different things so should be a good thing to give them a specified name, to get easily which of the 2 methods you are referring at.

thomasporro replied:
You're right! I will modify them

MrVlado reviewed (approved):
There are a lot of variables with settled values and I can't understand exactly what thay stand for and why you fixed that value for them. What does happen if I change them ? Can I do so ?

thomasporro reviewed (commented):
I will upload the changes in an hour

thomasporro committed:
Update ImageProcessing.java

-Modified the import statements according to the style doc
-Moved comment to explain what the commented methods do
-Modified the names and put one right after the other the two methods "conversion"
